homeassistant/components/tplink/config_flow.py

```diff
@@ -89,7 +97,16 @@ async def async_step_user(
             if not (host := user_input[CONF_HOST]):
                 return await self.async_step_pick_device()
             try:
-                device = await self._async_try_connect(host, raise_on_progress=False)
+                hSplit = host.split(":")
```

Comment:
How should the user know that they can enter port by using a colon in the host form? Normally the form name or description should inform the user what data should and can be entered. Do we have any precedence where the host form is used to enter both host and port?

Reply:
I updated the description for the host.

Reply:
We should also look at precedence.

Reply:
In the same commit, I updated variable names from hSplit to parts in two places.

Reply:
I was referring to this question:

> Do we have any precedence where the host form is used to enter both host and port?

Reply:
Yes, this is a well-known notation for specifying port in URLs:

https://en.wikipedia.org/wiki/Port_(computer_networking)#:~:text=Port%20numbers%20are%20sometimes%20seen,8080%20of%20the%20HTTP%20server.

Reply:
Right, but I mean in Home Assistant config flows.

Reply:
I am not aware of any precedence in Home Assistant. However in this case the use case for the user is exactly the same as in a browser. In a browser, there is a default port 80 and you don't have to specify it. Most non-dev people are not even aware of the possibility to specify a port in a browser. Similarly, in this case, most people will not use the port and it makes total sense not to put it in a separate field.

Reply:
I don't agree with you. Are you a gatekeeper for merging this PR? Can I disagree with you and still move forward with merging this PR?

Should we wait for @rytilahti to chime in on the subject? What if @rytilahti  does not agree with you either?

Reply:
Yes, it currently blocks this PR from being merged. Codeowners should also weigh in. If members disagree we'll need to find concensus among further members.

Reply:
I agree with Martin, this is not a good way to handle this input and also not the way this is commonly handled in Home Assistant either.

> The only form item in Home Assistant that I'm aware accepting both host and port is the url form. If that doesn't fit this case, I suggest using separate forms for host and port.

☝️ Exactly that.

Reply:
Hey just curious as a user of tplink, how would you be able to change the port that these devices use? I thought they were hard coded to use port 9999